android/java/org/chromium/chrome/browser/settings/VpnCalloutPreference.java

```diff
@@ -67,8 +68,12 @@ public void onBindViewHolder(PreferenceViewHolder holder) {
         });
         Button btnLearnMore = (Button) holder.findViewById(R.id.btn_learn_more);
         btnLearnMore.setOnClickListener(v -> {
-            BraveVpnUtils.openBraveVpnPlansActivity((Activity) getContext());
-            BraveVpnPrefUtils.setCallout(false);
+            if (!InternetConnection.isNetworkAvailable(getContext())) {
```

Comment:
seems like this should listen for internet connection status changes?

Reply:
@bridiver we need to check for the  connection only when user clicks on the option. We do check for the vpn specific connection in the callback